Source/WebCore/css/calc/CSSCalcOperationNode.cpp

```diff
@@ -823,14 +810,14 @@ void CSSCalcOperationNode::combineChildren()
         m_children.clear();
         m_children.append(WTFMove(newChild));
     }
-    if (isSteppedNode()) {
+    if (isSteppedNode() && canCombineAllChildren()) {
         auto combinedUnitType = m_children[0]->primitiveType();
         double resolvedValue = doubleValue(combinedUnitType);
         auto newChild = CSSCalcPrimitiveValueNode::create(CSSPrimitiveValue::create(resolvedValue, combinedUnitType));
         m_children.clear();
         m_children.append(WTFMove(newChild));
     }
-    if (isRoundOperation()) {
+    if (isRoundOperation() && canCombineAllChildren()) {
```

Comment:
Is there a reason this check isn't necessary for the rest of the calc functions?

Reply:
Not all calc functions end up in combineChildren() I think and a lot of calc functions only deal with primitive values.